thirdparty/mcap/mcap/writer.inl

```diff
@@ -53,7 +56,23 @@ void FileWriter::handleWrite(const std::byte* data, uint64_t size) {
   assert(file_);
   const size_t written = std::fwrite(data, 1, size, file_);
   (void)written;
-  assert(written == size);
+  if (written != size)
+  {
+#ifdef _WIN32
+    std::filesystem::space_info space = std::filesystem::space("C:\\");
+#else
+    std::filesystem::space_info space = std::filesystem::space("/");
+#endif
+
+    if (space.available < size)
+    {
+        logError(
+            DDSRECORDER_MCAP_HANDLER,
+            "Not enough space available in disk. Space available: " << space.available);
+
+        monitor_error("DISK_FULL");
```

Comment:
We should avoid to add unnecessary dependencies, and more important mutual dependencies. I think a better way would be to throw an exception that is captured in McapHandler.

Reply:
I still see an include, that is not being used. I guess it does not have to do with this PR though.